### ISSUE TITLE
DSub / Ultrasonic can browse Ampache Catalog, but can't play files

### ISSUE THREAD

**Issue by pollux105** [ampache/ampache#1751]
# Environment

- Raspberry Pi under Raspbian running Apache with PHP7 and Curl / mod_rewrite / all the needed modules enabled
- Music files (flac and mp3) on a NAS mounted on the raspberry with nfs (no writing permission, only reading)
- Ampache 3.8.8-release

 * Client type:
   - Web
   - SubSonic

Dsub ou Ultrasonic for Android (last versions)   

# Settings

Only one admin account at the moment, the same used locally, remotely and with Dsub/Ultrasonic.

# Description

I can perfectly browse my catalog and listen to song with the web interface.
I can perfectly browse my catalog with the android apps (Dsub / Ultrasonic).

Dsub and UltraSonic can't play any song, would it be encoded with flac or mp3 (I probably have a bad set-up for reencoding flac btw). It just indicate 0B download.

# Logs
ampache.log 

```
2018-07-23 12:42:41 [admin] (session) -> Session created: abd1c554d3c7b080851837b4e7cfcbab
2018-07-23 12:42:41 [admin] (subsonic) -> Stream proxy: http://ampache.xxxxxxx.eu:80/play/index.php?ssid=abd1c554d3c7b080851837b4e7cfcbab&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3
2018-07-23 12:42:41 [ampache] (play) -> Asked for type {song}
2018-07-23 12:42:41 [admin] (session) -> abd1c554d3c7b080851837b4e7cfcbab has been extended to Mon, 23 Jul 2018 14:42:41 +0000 extension length 7200
2018-07-23 12:42:41 [admin] (play) -> Playing file (/mnt/music/2000-2010/[ Fiona Apple ] - Extraordinary Machine/Fiona Apple - Not About Love.mp3}...
2018-07-23 12:42:41 [admin] (play) -> Media type {mp3}
2018-07-23 12:42:41 [admin] (play) -> Custom play action {}
2018-07-23 12:42:41 [admin] (play) -> Transcode to {}
2018-07-23 12:42:41 [admin] (play) -> Transcoding is not enabled for this media type. Valid types: {["native"]}
2018-07-23 12:42:41 [admin] (play) -> Starting stream of /mnt/music/2000-2010/[ Fiona Apple ] - Extraordinary Machine/Fiona Apple - Not About Love.mp3 with size 10518681
2018-07-23 12:42:41 [admin] (play) -> Registering stream stats for {Fiona Apple - Not About Love}...
2018-07-23 12:42:41 [admin] (user.class.php) -> Updating stats for {song/6276} {okhttp/3.10.0}...
2018-07-23 12:42:41 [admin] (user.class.php) -> Scrobbling explicitly skipped
2018-07-23 12:45:42 [admin] (play) -> Stream ended at 126976 (126976) bytes out of 10518681

```
access.ampache.log 
```
10.0.1.1 - - [23/Jul/2018:12:36:26 +0000] "GET /rest/getCoverArt.view?id=al-200000517&size=200&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 105902 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:33 +0000] "GET /rest/getArtist.view?id=100000283&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 1134 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:33 +0000] "GET /rest/getCoverArt.view?id=al-200000534&size=200&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 216916 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:34 +0000] "GET /rest/getAlbum.view?id=200000534&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 12551 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:35 +0000] "GET /rest/getCoverArt.view?id=200000534&size=512&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 216916 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:39 +0000] "GET /rest/getCoverArt.view?id=200000534&size=1920&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 216916 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:40 +0000] "GET /rest/getUser.view?username=admin&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 755 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:36:39 +0000] "GET /rest/stream.view?id=300006162&maxBitRate=0&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 200 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:05 +0000] "GET /rest/getArtist.view?id=100000285&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 1107 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:05 +0000] "GET /rest/getCoverArt.view?id=al-200000536&size=200&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 481418 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:07 +0000] "GET /rest/getAlbum.view?id=200000536&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 14158 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:17 +0000] "GET /rest/getArtist.view?id=100000291&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 1126 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:18 +0000] "GET /rest/getCoverArt.view?id=al-200000543&size=200&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 47410 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:19 +0000] "GET /rest/getAlbum.view?id=200000543&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 13012 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:19 +0000] "GET /rest/getCoverArt.view?id=200000543&size=512&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 47410 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:25 +0000] "GET /rest/getUser.view?username=admin&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 755 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:25 +0000] "GET /rest/getCoverArt.view?id=200000543&size=1920&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 47410 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:38:25 +0000] "GET /play/index.php?ssid=f5ce575075f69d61f29b52dcb54853e8&type=song&oid=6273&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Oh%20Well.mp3 HTTP/1.1" 200 123922 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:39:25 +0000] "GET /play/index.php?ssid=39fa08ded903725b6495afd4bdf2c47c&type=song&oid=6273&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Oh%20Well.mp3 HTTP/1.1" 200 116330 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:40:25 +0000] "GET /play/index.php?ssid=c77de7de27d24ae79c97df4cc11956d5&type=song&oid=6273&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Oh%20Well.mp3 HTTP/1.1" 200 116826 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:25 +0000] "GET /rest/getArtist.view?id=100000289&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 1120 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:26 +0000] "GET /rest/getCoverArt.view?id=al-200000540&size=200&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 21777 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:41:25 +0000] "GET /play/index.php?ssid=2722cddb8ba2333ff714584962089711&type=song&oid=6273&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Oh%20Well.mp3 HTTP/1.1" 200 125162 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:27 +0000] "GET /rest/getAlbum.view?id=200000540&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 17471 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:27 +0000] "GET /rest/getCoverArt.view?id=200000540&size=512&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 21777 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:35 +0000] "GET /rest/getArtist.view?id=100000291&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 1126 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:36 +0000] "GET /rest/getAlbum.view?id=200000543&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 13012 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:41 +0000] "GET /rest/getUser.view?username=admin&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 755 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:25 +0000] "GET /play/index.php?ssid=e2ae681913e7b38c76893c227eb8e8fa&type=song&oid=6273&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Oh%20Well.mp3 HTTP/1.1" 200 122058 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:42:41 +0000] "GET /play/index.php?ssid=abd1c554d3c7b080851837b4e7cfcbab&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 115490 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:43:41 +0000] "GET /play/index.php?ssid=8969c071a1325d51ba53b56f7a9cfb1c&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 120226 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:44:42 +0000] "GET /play/index.php?ssid=92855c731804b23dcdf30eb090e96d01&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 125770 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:45:42 +0000] "GET /play/index.php?ssid=485a10dab24c30d2394e7fc154e7fa08&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 120826 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:46:48 +0000] "GET /play/index.php?ssid=7efd2841fddee21ada384519668cd23f&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 124570 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:48:06 +0000] "GET /rest/stream.view?id=300006276&maxBitRate=0&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 200 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:48:43 +0000] "GET /play/index.php?ssid=d2217b3ce56869631f4129f9cdc573a8&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 HTTP/1.1" 200 120226 "-" "okhttp/3.10.0"
10.0.1.1 - - [23/Jul/2018:12:49:43 +0000] "GET /rest/stream.view?id=300006276&maxBitRate=0&u=admin&c=Ultrasonic&f=json&v=1.11.0&p=enc:6B6E6E6468786C31696D63696E796B77 HTTP/1.1" 200 200 "-" "okhttp/3.10.0"
```

And I get nor error in error.ampache.og

For info, if I try to copy/paste the following line in any browser, the song is download with no problem : `ampache.xxxxx.eu/play/index.php?ssid=7efd2841fddee21ada384519668cd23f&type=song&oid=6276&uid=1&client=Ultrasonic&noscrobble=1&player=api&name=Fiona%20Apple%20-%20Not%20About%20Love.mp3 `

**Comment by wagnered** [comment]
@pollux105 

Your ampache log shows http://ampache.xxxxxxx.eu.

Somehow this seems to be a redirection problem.

Could you include the "Path Vars" section of your config/ampache.cfg.php file?

Is the server url above a hosted server?

if the DSub client is on the same network as the Ampache server, please try point the server to the IP address of the Ampache server (ie: http://192.168.*.*) instead of the URL.

Also make sure play/.htaccess exists.

**Comment by pollux105** [comment]
xxxxxx stands for my domain name for confidentiality reasons :)

the config file show this on the path vars section :

```
;#########################################################
; Path Vars                                              #
;#########################################################

; The public http host of your server.
; If not set, retrieved automatically from client request.
; This setting is required for WebSocket server
; DEFAULT: ""
;http_host = "localhost"

; The public http port of your server.
; If not set, retrieved automatically from client request.
; DEFAULT:
;http_port = 80

; The public path to your ampache install
; Do not put a trailing / on this path
; For example if your site is located at http://localhost
; than you do not need to enter anything for the web_path
; if it is located at http://localhost/music you need to
; set web_path to /music
; DEFAULT: ""
web_path = ""

; The local http url of your server.
; If not set, retrieved automatically from server information.
; DEFAULT: ""
;local_web_path = "http://localhost/ampache"

```

The server is host at home on a raspberry pi... not sure i understand your question.
I specifically tried the Dsub client from another network as it could not solve the domain name from within.

And finally, play/.htaccess is the following : 
```
<IfModule mod_rewrite.c>
    RewriteEngine On
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-s
    RewriteRule ^art/([^/]+)/([^/]+)/([0-9]+)/thumb([0-9]*)\.([a-z]+)$ /image.php?object_type=$2&object_id=$3&auth=$1&thumb=$4&name=art.jpg [L]
    RewriteRule ^([^/]+)/([^/]+)/([^/]+)/([^/]+)(/.*)?$ /play/$5?$1=$2&$3=$4 [N,QSA]
    RewriteRule ^([^/]+)/([^/]+)(/.*)?$ /play/$3?$1=$2 [N,QSA]
    RewriteRule ^(/[^/]+|[^/]+/|/?)$ /play/index.php [L,QSA]
</IfModule>
```

I tried changing play/ with ampache/play but it does not seems to make any difference.
for ingo, my apache2 Vhost is set as following : 
```
<VirtualHost *:80>
        DocumentRoot "/var/www/html/ampache"
        ServerName ampache.xxx.eu
        <Directory "/var/www/html/ampache">
                Options +FollowSymLinks
                AllowOverride All
                Require all granted
        </Directory>
        ErrorLog /var/log/apache2/error.ampache.log
        CustomLog /var/log/apache2/access.ampache.log combined
</VirtualHost>
```

**Comment by linucksrox** [comment]
I'm having a similar problem, although everything worked OK on 3.8.8, and ran into this problem after upgrading to 3.8.9

```
2018-07-24 13:04:40 [eric] (subsonic) -> Stream proxy: https://music.example.com:443/play/ssid/d172757cc767f9551c231bd54c6c5caf/type/song/oid/3796/uid/2/client/DSub/noscrobble/1/bitrate/320/player/api/name/Modern%20Day%20Babylon%20-%20Sleeper.mp3
2018-07-24 13:04:40 [ampache] (play) -> Asked for type {song}
2018-07-24 13:04:40 [eric] (session) -> d172757cc767f9551c231bd54c6c5caf has been extended to Tue, 24 Jul 2018 17:04:40 -0400 extension length 14400
2018-07-24 13:04:40 [eric] (play) -> Playing file (/mnt/freenas/smbshare/music/Public/Modern Day Babylon/Coma/01 Sleeper.flac}...
2018-07-24 13:04:40 [eric] (play) -> Media type {flac}
2018-07-24 13:04:40 [eric] (play) -> Custom play action {}
2018-07-24 13:04:40 [eric] (play) -> Transcode to {}
2018-07-24 13:04:40 [eric] (subsonic) -> Stream error: The requested URL returned error: 500 Internal Server Error

I'm guessing there's some problem detecting the transcode type judging by the line "Transcode to {}"?
```

**Comment by wagnered** [comment]
Could you try setting transcoding to "Never" in admin->Server Config->Streaming? It is not needed for subsonic clients and may eliminate some possibilities.

if DSub is on the same local network as the Ampache server, Setting DSub server to the IP address of the Ampache server should eliminate redirect problems.  If you don't know the IP address, from a command line, type in `ip addr`. and look for the inet entry.  Don't use the one for 'lo' that's the localhost (127.0.0.1).  In the dsub server address type in the ip for the url(example: http://192.168.2.4). You might have to take DSub off line for a moment or clear the cache before it tries using the new address.

**Comment by pollux105** [comment]
I did put "never" in streaming conf (btw, where is this setting stored ? ampache.cfg.php did not seems to have changed ?).
On "lan", it works ! and I don't know why, I have to solve this (I don't want to use a VPN :D )

**Comment by wagnered** [comment]
it's stored in the database.
Do you have a router  between the internet and the Ampache server? If so, is port forwarding set up correctly? 

This issue seems to be a duplicate of #1639(closed).  Maybe some comments in that issue may help.

The closest I can come to replicating your setup is to enter my DDNS URL as the DSub server address so it is forced to access my Ampache server through the internet. DSub works perfectly this way on Ampache 3.8.8/3.8.9

**Comment by pollux105** [comment]
I'm still trying to find out why is not working. It's clearly a duplicate of #1639 but I could not find out any solution that could be successfully applied to my issue. 
@simpledisaster, who had a very similar issue, solved the problem by reinstalling everything from scratch then it worked... that does not help me much 🤣 

So I went for a new install of Ampache (erasing database as well) and going for 3.8.9

I was doing reverse proxy through my NAS : 
*:80 >> router NAT >> 10.0.1.1:80 (NAS with reverse proxy for ampache.mydomain.eu) >> 10.0.1.2/var/www/ampache

I am now routing directly the port 80 to my raspPi. 
*:80 >> router NAT >> 10.0.1.2/var/www/ampache

But still, I have the same issue.

I tried to change the "type of playback" option, but it really affects only the web browser behaviour.
What's surprising is that when I just copy the url from the ampache log in any web browser, the download is ok.

ampache.log : 
```
2018-07-26 14:48:26 [admin] (session) -> Session created: 897dee9d9a2e2a551237ef29e15019bd
2018-07-26 14:48:26 [admin] (subsonic) -> Stream proxy: http://ampache.dutriez.eu:80/play/index.php?ssid=897dee9d9a2e2a551237ef29e15019bd&type=song&oid=76&uid=1&client=DSub&noscrobble=1&bitrate=166&player=api&name=Antony%20and%20the%20Johnsons%20-%20For%20Today%20I%20Am%20a%20Boy.mp3
2018-07-26 14:48:26 [ampache] (play) -> Asked for type {song}
2018-07-26 14:48:26 [admin] (session) -> 897dee9d9a2e2a551237ef29e15019bd has been extended to Thu, 26 Jul 2018 16:48:26 +0000 extension length 7200
2018-07-26 14:48:26 [admin] (play) -> Playing file (/mnt/music/2000-2010/[ Antony and The Johnsons ] - I Am A Bird Now/Antony And The Johnsons - For Today I Am A Boy.mp3}...
2018-07-26 14:48:26 [admin] (play) -> Media type {mp3}
2018-07-26 14:48:26 [admin] (play) -> Custom play action {}
2018-07-26 14:48:26 [admin] (play) -> Transcode to {}
2018-07-26 14:48:26 [admin] (play) -> Transcode disabled in user settings.
2018-07-26 14:48:26 [admin] (play) -> Starting stream of /mnt/music/2000-2010/[ Antony and The Johnsons ] - I Am A Bird Now/Antony And The Johnsons - For Today I Am A Boy.mp3 with size 3314015
2018-07-26 14:48:26 [admin] (play) -> Registering stream stats for {Antony and the Johnsons - For Today I Am a Boy}...
2018-07-26 14:48:26 [admin] (user.class.php) -> Updating stats for {song/76} {DSub}...
2018-07-26 14:48:26 [admin] (user.class.php) -> Scrobbling explicitly skipped
2018-07-26 14:48:56 [admin] (session) -> Session created: 432f9cee088ecc7ad5cbdf3e70f272fc
2018-07-26 14:48:56 [admin] (subsonic) -> Stream proxy: http://ampache.dutriez.eu:80/play/index.php?ssid=432f9cee088ecc7ad5cbdf3e70f272fc&type=song&oid=76&uid=1&client=DSub&noscrobble=1&bitrate=166&player=api&name=Antony%20and%20the%20Johnsons%20-%20For%20Today%20I%20Am%20a%20Boy.mp3
2018-07-26 14:48:56 [ampache] (play) -> Asked for type {song}
2018-07-26 14:48:56 [admin] (session) -> 432f9cee088ecc7ad5cbdf3e70f272fc has been extended to Thu, 26 Jul 2018 16:48:56 +0000 extension length 7200
2018-07-26 14:48:56 [admin] (play) -> Playing file (/mnt/music/2000-2010/[ Antony and The Johnsons ] - I Am A Bird Now/Antony And The Johnsons - For Today I Am A Boy.mp3}...
2018-07-26 14:48:56 [admin] (play) -> Media type {mp3}
2018-07-26 14:48:56 [admin] (play) -> Custom play action {}
2018-07-26 14:48:56 [admin] (play) -> Transcode to {}
2018-07-26 14:48:56 [admin] (play) -> Transcode disabled in user settings.
2018-07-26 14:48:56 [admin] (play) -> Starting stream of /mnt/music/2000-2010/[ Antony and The Johnsons ] - I Am A Bird Now/Antony And The Johnsons - For Today I Am A Boy.mp3 with size 3314015
2018-07-26 14:48:56 [admin] (play) -> Registering stream stats for {Antony and the Johnsons - For Today I Am a Boy}...
2018-07-26 14:48:56 [admin] (user.class.php) -> Updating stats for {song/76} {DSub}...
2018-07-26 14:48:56 [admin] (user.class.php) -> Scrobbling explicitly skipped
```

We see that every 30 second, Dsub tries again to get the song, but nothing happens.

**Comment by wagnered** [comment]
> and a few minutes later :

I just checked my server response time and there was less than 10 seconds to end streaming of a 15 mb song  via my ddns URL.

The " Stream ended at" line should be only a line or two after: "Updating stats for..." line.

Could you include your full ampache.cfg.php file?  I'm very sure that it is a problem with routing, or maybe a firewall problem with the router or NAS, but the config file might reveal something. 

Also, could you try using the ip address of ampache.dutriez.eu instead of the URL?

I can't replicate your issue using http i'm just giving you possibilities.  Sorry...

**Comment by pollux105** [comment]
I've just tried this : 

1. Start playing a song with DSub/Ultrasonic (stream ended after a few bytes... but always not so few, about 20000 to 60000, which is kind of strange...) 
2. Copying/Pasting the stream proxy url from the Ampache log to the same phone I use Dsub/UltraSonic on my phone browser. 

Conclusion : the song is downloaded and played with my phone default player (VLC here).

From the Ampache log, stream ends at the end of the song.
Which tends to mean that this is not a network problem, rather the way Ampache/Subsonic API/DSub is considering the stream... strange it works on local network.

I really don't understand what happens. Streaming to Dsub or "kind of downloading a song" with a browser is different ?

**Comment by wagnered** [comment]
The problem isn't with your modem.

You might try setting Admin->System-config->Streaming->Enable url rewriting to enable and also in the user configuration.

But I'm convinced that it is a problem with router settings.

I suggest that you install and get familiar with wireshark to watch the  HTTP/TCP conversation between  the server and ampache.dutriez.eu using it's ip address.

![wireshark](https://user-images.githubusercontent.com/6115438/43333088-9bbf2ec2-917e-11e8-962d-42baf7850c07.png)

Notice that a continuous series of ACKs are sent between my server (192.168.2.3) and the DSub client which is at 73.42.226.251 (my DDNS),   If the client isn't sending ACKs, it probably means that the router isn't forwarding the server response to the client.

**Comment by simpledisaster** [comment]
@pollux105 

I'm just gonna chime in here with a couple of "make sures" and mention a couple of things that I did without really knowing which one of these exactly helped for it to work as it should. They were not mentioned in any tutorials or instructions, but in the end they did make a difference for me.

- Make sure that there really is a stable connection between your Ampache server and your remote host. I'm with no-ip. Even though I thought everything was set up accordingly and a connection was established, these two didn't communicate that well yet (e.g. port forwarding must be confirmed). Go through the right setup process a couple of times just to be sure.

- Make sure that your webserver is listening appropriately. I set my Ampache "device" to a static IP. I included to my apach2.config that it would listen to 

```
Listen 192.168.2.XXX:80 
Listen 192.168.2.XXX:8000 
Listen 127.0.0.1:80 
Listen 127.0.0.1:8000
```

just to be sure.

Include ports.conf, in which you should have your ports written down that are open for forwarding.

- And here the important parts from my ampache.cfg.php

```
;#########################################################
; Path Vars                                              #
;#########################################################

; The public http host of your server.
; If not set, retrieved automatically from client request.
; This setting is required for WebSocket server
; DEFAULT: ""
;http_host = "localhost"

; The public http port of your server.
; If not set, retrieved automatically from client request.
; DEFAULT:
http_port = 80

; The public path to your ampache install
; Do not put a trailing / on this path
; For example if your site is located at http://localhost
; than you do not need to enter anything for the web_path
; if it is located at http://localhost/music you need to
; set web_path to /music
; DEFAULT: ""
web_path = "/ampache"

; The local http url of your server.
; If not set, retrieved automatically from server information.
; DEFAULT: ""
local_web_path = "http://localhost/ampache"

```
Watch out for the ; (semicolons) in this one.

- Make sure that in your client-app of choice you insert your remote address as follows:

`http://yourserverremoteaddress:port/ampache`

For me this is /ampache at the end, because I have my Ampache folder with all contents in the root webserver folder (so /www/ampache).

- I see you got .htaccess and all the other important factors down that are mentioned in the Wiki. 

- Make sure that all of your necessary services (e.g. remote services, apache2, etc) are running. Especially when trying to find a solution for a problem, it ends up in a lot of experimenting and rebooting. It might happen along the way that one of these is actually not running.

- And once again, make sure that you covered all necessary parts that are mentioned in the Wiki and other in other instructions that you can find online

These are a couple of things I remember along the way and maybe a couple of things you can check again or that give you a hint for further debugging. Hope it helps!

**Comment by pollux105** [comment]
@simpledisaster 

could you include your apache2 Vhost related to Ampache ?

On my side, I have tried another modem/router combo (pfsense router) and same result. But I still have hope :)

**Comment by wagnered** [comment]
@pollux105 
I apologize for not getting back to you sooner. I've been very busy as the only active developer.

Has this issue been resolved?

**Comment by pollux105** [comment]
Hi, 
No, it has not, I went for another music server that worked instantly with the same network environment... but I would be happy to get a solution for Ampache... Since running a tomcat java server for a simple music server is overkill in my opinion.

**Comment by pollux105** [comment]
I will check this as soon as possible :)

**Comment by pollux105** [comment]
Better later than never.

I reinstalled from scratch. version 3.9.0.
And it works perfectly now.... I don't see any reason why it would not before :/

Just one thing, on Dsub, it does not start playing the song before it's fully loaded, I don't know if this is standard behaviour.

**Comment by dubwoc** [comment]
Upgrading to 3.9.0 fixed this problem entirely web and dsub work without modification.

@lachlan-00: I suggest the issue be closed unless others are still dealing with problems.

**Comment by lachlan-00** [comment]
Thanks @dubwoc I'll have a read of both issues and close them tonight when I get back from work

**Comment by lachlan-00** [comment]
Closing fixed in v3.9.0
